nemo/collections/llm/recipes/log/default.py

```diff
@@ -67,8 +71,37 @@ def default_log(
 
 @cli.factory(is_target_default=True)
 def default_resume(resume_if_exists=True, resume_ignore_no_checkpoint=True) -> Config[nl.AutoResume]:
+    """Factory function to configure AutoResume."""
     return Config(
         nl.AutoResume,
         resume_if_exists=resume_if_exists,
         resume_ignore_no_checkpoint=resume_ignore_no_checkpoint,
     )
+
+
+def get_global_step_from_global_checkpoint_path(path: Union[str, Path]) -> int:
```

Comment:
Why is this in recipes? Looks like a util function.

Reply:
this file defines the default filename convention in L59, so i've added this util function here as a convenience. 
since the filename is still ultimately configurable by the user, there's no generic solution we can provide unless we read the checkpoint data and get the step

Reply:
It doesn't seem to be used anywhere.

Reply:
it's a utility function that users can pass to `update_trainer_local_checkpoint_io` if they use the filename format.